components/supersaas/sources/changed-appointments.js

```diff
@@ -0,0 +1,39 @@
+const supersaas = require('../supersaas.app.js');
+
+module.exports = {
+  key: 'supersaas-changed-appointments',
+  name: 'New or changed appointments',
+  description: `This event source tracks new and changed appointment events from the selected schedules and emits them.`,
+  version: '0.0.1',
+
+  props: {
+    supersaas,
+    schedules: { propDefinition: [supersaas, 'schedules'] },
+    db: "$.service.db",
+    http: '$.interface.http',
+  },
+
+  hooks: {
+    async activate() {
+      const { http, schedules } = this;
+
+      this.db.set('activeHooks', await this.supersaas.createHooks(schedules.map(x => ({
+        event: 'C', // change_appointment
+        parent_id: Number(x),
+        target_url: http.endpoint,
+      }))));
+    },
+
+    async deactivate() {
+      await this.supersaas.destroyHooks(this.db.get('activeHooks') || []);
+      this.db.set('activeHooks', []);
+    },
+  },
+
+  async run(ev) {
+    // TODO: Authenticate requests? (see GitHub's HMAC signature approach)
```

Comment:
Do you pass a signature or other secret on webhook requests? If not, the validation isn't necessary from our perspective - we always include it if the source service allows, but not required.

Reply:
Nope, that's what I was thinking of adding. I'm only worried of endpoint name conflicts causing servers from other Pipedream users to send payload I'm not interested in and causing errors in our own endpoints. Something to prevent endpoint guess attacks might also be nice, but if you think it may be overkill, I'll trust you anytime :P

Reply:
Gotcha. Based on our algorithm for generating endpoints the probability of collisions is infinitesimal, and we should also block that from happening at the API level.